File: CS/WindowsFormsApplication1/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraTreeList.Columns;
using DevExpress.XtraTreeList.Nodes;

namespace WindowsFormsApplication1 {
    public partial class Form1 : Form {
        public Form1() {
            InitializeComponent();
        }

        private void simpleButton1_Click(object sender, EventArgs e) {
            treeList1.ExportToXmlEx("treeListData.xml");
        }

        private void simpleButton2_Click(object sender, EventArgs e) {
            if(System.IO.File.Exists("treeListData.xml")) {
                treeList1.ImportFromXmlEx("treeListData.xml");
            }
        }

        private void Form1_Load(object sender, EventArgs e) {
            string[] fields = new string[] { "FirstName", "LastName", "Address", "Certificate" };                
            for (int i = 0; i < fields.Length; i++)
			{
                TreeListColumn treeListColumn = treeList1.Columns.Add();
                treeListColumn.FieldName = fields[i];
                treeListColumn.VisibleIndex = i;			 
			}

            TreeListNode nodeBob = treeList1.AppendNode(new object[] { "Bob", "Brown", new CustomerAddress() { City = "Chicago", Country = "USA" }, new CustomerCertificate() { Series = "AAA", Number = "11-B-0235" } }, null);
            TreeListNode nodeJohn = treeList1.AppendNode(new object[] { "John", "Stone", new CustomerAddress() { City = "Toronto", Country = "Canada" }, new CustomerCertificate() { Series = "BBB", Number = "11-B-0235" } }, nodeBob);
            TreeListNode nodeJames = treeList1.AppendNode(new object[] { "James", "Carrick", new CustomerAddress() { City = "New York", Country = "USA" }, new CustomerCertificate() { Series = "CCC", Number = "11-B-0235" } }, nodeBob);
            TreeListNode nodeSara = treeList1.AppendNode(new object[] { "Sara", "Taylor", new CustomerAddress() { City = "Vancouver", Country = "Canada" }, new CustomerCertificate() { Series = "DDD", Number = "11-B-0235" } }, null);
            TreeListNode nodeMarry = treeList1.AppendNode(new object[] { "Marry", "Smith", new CustomerAddress() { City = "Calgary", Country = "Canada" }, new CustomerCertificate() { Series = "EE", Number = "11-B-0235" } }, nodeSara);
            TreeListNode nodeMonika = treeList1.AppendNode(new object[] { "Monika", "York", new CustomerAddress() { City = "Philadelphia", Country = "USA" }, new CustomerCertificate() { Series = "FFF", Number = "11-B-0235" } }, nodeSara);


        }
    }

    public class CustomerAddress {
        public string Country { get; set; }
        public string City { get; set; }

        public override string ToString() {
            return String.Format("{0} - {1}", Country, City);
        }
    }

    public class CustomerCertificate {
        public string Series { get; set; }
        public string Number { get; set; }

        public override string ToString() {
            return String.Format("{0} - {1}", Series, Number);
        }
    }
}

File: VB/WindowsFormsApplication1/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Linq
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraTreeList.Columns
Imports DevExpress.XtraTreeList.Nodes

Namespace WindowsFormsApplication1
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub simpleButton1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles simpleButton1.Click
            treeList1.ExportToXmlEx("treeListData.xml")
		End Sub

		Private Sub simpleButton2_Click(ByVal sender As Object, ByVal e As EventArgs) Handles simpleButton2.Click
			If System.IO.File.Exists("treeListData.xml") Then
				treeList1.ImportFromXmlEx("treeListData.xml")
			End If
		End Sub

		Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
			Dim fields() As String = { "FirstName", "LastName", "Address", "Certificate" }
			For i As Integer = 0 To fields.Length - 1
				Dim treeListColumn As TreeListColumn = treeList1.Columns.Add()
				treeListColumn.FieldName = fields(i)
				treeListColumn.VisibleIndex = i
			Next i

            Dim nodeBob As TreeListNode = treeList1.AppendNode(New Object() {"Bob", "Brown", New CustomerAddress() With {.City = "Chicago", .Country = "USA"}, New CustomerCertificate() With {.Series = "AAA", .Number = "11-B-0235"}}, -1)
			Dim nodeJohn As TreeListNode = treeList1.AppendNode(New Object() { "John", "Stone", New CustomerAddress() With {.City = "Toronto", .Country = "Canada"}, New CustomerCertificate() With {.Series = "BBB", .Number = "11-B-0235"} }, nodeBob)
			Dim nodeJames As TreeListNode = treeList1.AppendNode(New Object() { "James", "Carrick", New CustomerAddress() With {.City = "New York", .Country = "USA"}, New CustomerCertificate() With {.Series = "CCC", .Number = "11-B-0235"} }, nodeBob)
            Dim nodeSara As TreeListNode = treeList1.AppendNode(New Object() {"Sara", "Taylor", New CustomerAddress() With {.City = "Vancouver", .Country = "Canada"}, New CustomerCertificate() With {.Series = "DDD", .Number = "11-B-0235"}}, -1)
			Dim nodeMarry As TreeListNode = treeList1.AppendNode(New Object() { "Marry", "Smith", New CustomerAddress() With {.City = "Calgary", .Country = "Canada"}, New CustomerCertificate() With {.Series = "EE", .Number = "11-B-0235"} }, nodeSara)
			Dim nodeMonika As TreeListNode = treeList1.AppendNode(New Object() { "Monika", "York", New CustomerAddress() With {.City = "Philadelphia", .Country = "USA"}, New CustomerCertificate() With {.Series = "FFF", .Number = "11-B-0235"} }, nodeSara)


		End Sub
	End Class

	Public Class CustomerAddress
		Private privateCountry As String
		Public Property Country() As String
			Get
				Return privateCountry
			End Get
			Set(ByVal value As String)
				privateCountry = value
			End Set
		End Property
		Private privateCity As String
		Public Property City() As String
			Get
				Return privateCity
			End Get
			Set(ByVal value As String)
				privateCity = value
			End Set
		End Property

		Public Overrides Function ToString() As String
			Return String.Format("{0} - {1}", Country, City)
		End Function
	End Class

	Public Class CustomerCertificate
		Private privateSeries As String
		Public Property Series() As String
			Get
				Return privateSeries
			End Get
			Set(ByVal value As String)
				privateSeries = value
			End Set
		End Property
		Private privateNumber As String
		Public Property Number() As String
			Get
				Return privateNumber
			End Get
			Set(ByVal value As String)
				privateNumber = value
			End Set
		End Property

		Public Overrides Function ToString() As String
			Return String.Format("{0} - {1}", Series, Number)
		End Function
	End Class
End Namespace
